modules/tinymce/src/plugins/link/main/ts/core/Utils.ts

```diff
@@ -60,11 +61,38 @@ const applyRelTargetRules = (rel: string, isUnsafe: boolean): string => {
 
 const trimCaretContainers = (text: string): string => text.replace(/\uFEFF/g, '');
 
+const isParentAnchor = (element: SugarElement<Node>): boolean => Traverse.parent(element).exists(SugarNode.isTag('a'));
+const isNextSiblingAnchor = (element: SugarElement<Node>): boolean => Traverse.nextSibling(element).exists(SugarNode.isTag('a'));
+const getTextParagraphNode = (start: SugarElement<Node>, end: SugarElement<Node>): SugarElement<Node> | null => {
+  if (SugarNode.isText(start) && SugarNode.isElement(end)) {
+    return start;
+  }
+  if (SugarNode.isText(end) && SugarNode.isElement(start)) {
+    return end;
+  }
+  return null;
+};
+const isTexParagraphtNode = (start: SugarElement<Node>, end: SugarElement<Node>): boolean => getTextParagraphNode(start, end) !== null;
+
 const getAnchorElement = (editor: Editor, selectedElm?: Element): HTMLAnchorElement | null => {
   selectedElm = selectedElm || editor.selection.getNode();
+  const rng = editor.selection.getRng();
+  const start = SugarElement.fromDom(rng.startContainer);
```

Comment:
There is a lot of details here fell free to reach out to me on slack if anything is unclear:

What you want to get here is the BR element. Dom ranges has two modes one is text node based offsets then other one is element based offsets. For this case the endContainer would be the parent paragraph element and the endOffset would be set just before that BR. So for a selection that is `<p><a href="#">[abc</a>]<br></p>` the startContainer would be the `abc` test and the endContainer would be the `P` element and the endOffset would be 1 since it's the index before the start.

So you could resolve that index by doing something like this (dirty pseudo code):
```
if (rng.endContainer.nodeType === 1) {
  return rng.endContainer.childNodes[rng.endOffset];
}
```

So if that is a BR you can then just do a resolvedNode.previousSibling and check if that sibling is the same as the startLink. I don't think you would need to use sugar here since it's pretty simple.

So if this:
editor.dom.getParent(tinymce.activeEditor.selection.getNode(), 'a[href]')

Doesn't return link but this does:
editor.dom.getParent(tinymce.activeEditor.selection.getStart(), 'a[href]')

Then you can resolve the end and see if that is a BR and if that BR has a previous sibling that is the same link element as the start.

This would only solve the simple case not wrapped links for that it would have to traverse. You could then use the tinymce TreeWalker then walk backwards from the resolved endNode to the startNode if you only find inline elements that are not short ended you can continue walking until you hit the link. That means it will not walk over text nodes or IMG elements or other BR elements etc. Checking if a element is a inline could be done using the schema api. So you do something like `editor.schema.getShortEndedElements().hasOwnProperty(traverseNode.nodeName)` to check if it's a IMG or BR then `!editor.schema.getBlockElements().hasOwnProperty(traverseNode.nodeName)` to check if it's a inline if it is not a block then it is an inline unless it's a text node.

So it would be something like in pseudo code:
`const isValidNode = (node) => isElement(node) && isInline(node) && !isShortEnded(node) && node !== startLink`

Also I think this function could be created and tested in isolation then you can feed it different content scenario cases and validate that it all returns the link expected. It would require a the schema and range so maybe something like:
const getSelectedLink = (editor, schema, range): Element? then you could setup a editor in the test set the content to something, feed it a range and see if you get back the expected element.

So maybe pull this out into a separate module called SelectedLink.ts or something add tests for that like SelectedLinkTest.ts that calls the above function on various content pieces. Then you add a specific integration test where you open the dialog etc with a range that is like the BR case to validate that the whole solution works.